src/sentry/sentry_apps/api/bases/sentryapps.py

```diff
@@ -123,7 +143,7 @@ def _get_organization_slug(self, request: Request):
         organization_slug = request.data.get("organization")
         if not organization_slug or not isinstance(organization_slug, str):
             error_message = "Please provide a valid value for the 'organization' field."
-            raise ValidationError({"organization": to_single_line_str(error_message)})
+            raise SentryAppError(ResourceDoesNotExist(error_message))
```

Comment:
What's the benefit of wrapping exceptions within these `SentryAppError`s? If we're crafting the error messages ourselves, could we just pass the message and status codes instead?

Reply:
Benefit of wrapping vs making everything SentryAppXError is just giving the clarity of what the 'causation error' is. Albeit we aren't sending stack traces for these types of errors so I'm not sure if people would ever see the wrapped error. Main benefit would be more info for the person reading the code/trace.

Reply:
That's fair, but I think the error message itself + status would carry enough context already. If we want to get this granular, maybe we could create different subclasses of `SentryAppError` that are more explicit and encode status codes for us? I'm just not used to seeing this pattern of composed exceptions in our codebase.

Reply:
we would have to catch all of these different errors in the custom exception handling if we don't wrap them, i suppose the wrappers all return 400 so we could just catch these directly?

Reply:
makes sense, I wasn't sure if just the error message was enough detail/context. There is that various subclass exceptions in the codebase with the built in rest_framework exception class (i.e APIException) but that's been a headache to deal with so I'd rather keep it simple for now ;-;

Reply:
> I wasn't sure if just the error message was enough detail/context.
That's a valid concern, and there wasn't anything wrong with your approach as far as I could tell. I just don't know the full implications of wrapping exceptions in exceptions like this 😅 

Regardless, Cathy's approach sounds good to me.

Reply:
💀 I didn't see Cathy's comment before right now/pushing